### DISCUSSION TITLE
Cannot add Git repo after changing the argocd service from LB to ClusterIP

### DISCUSSION THREAD

**Discussion by vikram-yerneni** [argoproj/argo-cd#15877] [Q&A]
I got an interesting challenge after removing the external load-balancer for the argocd server setup .

- Basically we removed the load-balancer for the argocd server and made it clusterip
- We use istio for the ingress routing, so we added a virtual service and assigned it to the  argocd server:port 80 using istio
- When using the virtual-service url, I can access the argo UI via browser and its all fine.
- Since we are using a virtual service, I can access the argocd cli and its fine.

The issue comes when I try to do a task where argocd server reaches out to other argocd services.
For example, when trying to add a repo via cli (using the virtual service), its failing with the below error:
` time="2023-10-06T13:13:46-04:00" level=fatal msg="rpc error: code = Unavailable desc = connection error: desc = \"transport: authentication handshake failed: read tcp argocd-server:52984->argocd-repo-service:8081: read: connection reset by peer\""`

Basically the traffic is coming in via a virtual service, the authentication is failing between argocd-server to argocd-repo-service 

Also I cant do a simple argocd appset list.  Same error as above.
Any clue on how to fix the authentication issue between multiple Argo services
